### ISSUE TITLE
Yarn workspaces support (mono-repo)

### ISSUE THREAD

**Issue by sumanbh** [Dwlad90/stylex-swc-plugin#6]
I tried the plugin on a mono-repo that uses yarn workspaces but I've been unsuccessful so far with what looks to be file resolution error:

```
Import trace for requested module:
../stylex-lib/index.js
./app/components/ButtonsDemo.tsx
thread '<unnamed>' panicked at src/shared/utils/common.rs:538:8:
called `Result::unwrap()` on an `Err` value: StripPrefixError(())
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '' panicked at /Users/geist/.cargo/registry/src/index.crates.io-6f17d22bba15001f/swc-0.273.24/src/plugin.rs:166:14:
failed to invoke plugin: failed to invoke plugin on 'Some("/nextjs-app-dir-stylex/packages/stylex-lib/index.js")'
```

### Steps to reproduce

https://github.com/sumanbh/nextjs-stylex-workspaces-issue

Note: I'm also setting [nmHoistingLimits](https://github.com/sumanbh/nextjs-stylex-workspaces-issue/blob/main/.yarnrc.yml#L4) otherwise the plugin wouldn't resolve the workspace lib `stylex-lib`.

**Comment by Dwlad90** [comment]
Hi @sumanbh,
Thank you for finding the problem.

I've tested your example with the original [`babel plugin`](https://www.npmjs.com/package/@stylexjs/babel-plugin) and the code with defining colors also doesn't work, although the following error is more informative: `Only static values are allowed inside of a stylex.create() call.`. This happens because the constants must be imported from a file with a relative path that ends with ".stylex.js", but in your example with importing colors from another workspace, the path is absolute.
```tsx
import { colors } from 'stylex-lib/colors.stylex';

// ...

<ThemeableButton
  onClick={onClick}
  style={(styles.bordered, testStyles.base)}
>
  Bordered Button
</ThemeableButton>

```
 Here is a [link](https://stylexjs.com/docs/api/javascript/defineVars/) to the official documentation.

In this project, I follow the API of the original plugin and even after fixing this code, it will throw an error, as mentioned above.

Perhaps you should open an issue in the original [StyleX](https://github.com/facebook/stylex/issues) repository, and if a solution is found, or has already been found, then I will happily implement it in this project as well.

*P.S.*
If you change all absolute paths to relatives, for example `../../../stylex-lib/colors.stylex` in the `packages/next-app/app/components/ButtonTokens.stylex.ts` file, then everything starts working as expected with the babel plugin:
```tsx
import { colors } from '../../../stylex-lib/colors.stylex';

// ...

export const buttonTokens = stylex.defineVars({
  bgColor: colors.blue12,
  textColor: 'white',
  cornerRadius: '4px',
  paddingBlock: '4px',
  paddingInline: '8px',
});
```

Unfortunately, there is still a problem with the SWC plugin, which also will be fixed soon in the next version.

**Comment by Dwlad90** [comment]
Hi @sumanbh,
Thank you for finding the problem.

The reason for the path resolving issues is how the plugin handles importing styles from an external file, which are passed to the `style` property:

```tsx
import * as stylex from '@stylexjs/stylex';
import { testStyles } from 'stylex-lib';

// ...

<ThemeableButton
  onClick={onClick}
  style={(styles.bordered, testStyles.base /* <-- this one */)}
>
  Bordered Button
</ThemeableButton>

// ...
```

If you remove them, all other styles are applied correctly.

```tsx
import * as stylex from '@stylexjs/stylex';

// ...

<ThemeableButton
  onClick={onClick}
  style={(styles.bordered)}
>
  Bordered Button
</ThemeableButton>

// ...
```

This issue will be fixed in the next version soon.

**Comment by Dwlad90** [comment]
Hi @sumanbh,

I have released `@stylexswc/swc-plugin@0.2.0` with a fix for this issue and prepared a [sandbox](https://codesandbox.io/p/github/sumanbh/nextjs-stylex-workspaces-issue/csb-rfvg8z/draft/recursing-elgamal) with the updated version of the plugin.

Could you check that everything works correctly now?
